code/game/objects/items/eightball.dm

```diff
@@ -186,12 +188,19 @@
 
 	voted.Cut()
 
-	return top_vote
+	var/list/top_options = haunted_answers[top_vote]
+	return pick(top_options)
+
+// Only ghosts can interact because only ghosts can open the ui
+/obj/item/toy/eightball/haunted/can_interact(mob/living/user)
+	return isobserver(user)
```

Comment:
As in the issue report linked mentions, `can_interact` failed for ghosts due to adjacency unless admin Ai interact was on